appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/EntityManagerWrapper.java

```diff
@@ -227,7 +227,9 @@ private EntityManager _getDelegate() {
                     }
                 }
             }
-
+            if (extendedEntityManager == null) {
+                throw new IllegalStateException("No EntityManager available in current extended persistence context.");
```

Comment:
The `delegate` set to the `extendedEntityManager` in this branch is expected to never return null (inferred from users of the method) but currently the `extendedEntityManager` could be null at this point which would lead to calling methods fail with a NPE because of the `delegate` being null. In such cases we not get the `IllegalStateException` instead. So we still get an exception for those cases where this is true but now its a more descriptive one at the point of the root cause and not a NPE later one.